pkg/operator/ceph/object/shared_pools.go

```diff
@@ -165,12 +180,10 @@ func toZonePlacementPools(spec cephv1.ObjectSharedPoolsSpec, ns string) map[stri
 	hasDefault := false
 	res := make(map[string]ZonePlacementPool, len(spec.PoolPlacements)+1)
 	for _, pp := range spec.PoolPlacements {
-		name := pp.Name
-		if pp.Name == defaultPlacementName {
+		if pp.Default {
 			hasDefault = true
-			name = defaultPlacementCephConfigName
 		}
-		res[name] = toZonePlacementPool(pp, ns)
+		res[pp.Name] = toZonePlacementPool(pp, ns)
```

Comment:
Similar concern/question here: if default bool is unset, should we use previous behavior?